File: www/python/add.py
#!/usr/bin/env python
import os
from urllib.parse import parse_qs

# get the values of num1 and num2 from the query string
query_string = os.environ.get('QUERY_STRING', '')
query_params = parse_qs(query_string)
num1 = int(query_params.get('num1', [''])[0])
num2 = int(query_params.get('num2', [''])[0])

# calculate the sum
sum = num1 + num2

# output the result
print("Content-Type: text/html")
print("")
print("<html><body>")
print("<h1>Sum of %d and %d is %d</h1>" % (num1, num2, sum))
print("</body></html>")

File: www/php/add.php
<?php
// Read input from query string (GET) or from standard input (POST)
if ($_SERVER['REQUEST_METHOD'] === 'GET') {
    // Read input from query string
    if (isset($_GET['num1']) && isset($_GET['num2'])) {
        $num1 = $_GET['num1'];
        $num2 = $_GET['num2'];
    } else {
        // Handle missing input
        die("Error: num1 and num2 must be provided in the query string.");
    }
} elseif ($_SERVER['REQUEST_METHOD'] === 'POST') {
    // Read input from standard input (POST)
    $input = file_get_contents('php://input');
    $inputArr = explode('&', $input);
    if (count($inputArr) === 2) {
        $num1 = explode('=', $inputArr[0])[1];
        $num2 = explode('=', $inputArr[1])[1];
    } else {
        // Handle missing input
        die("Error: num1 and num2 must be provided in the request body.");
    }
} else {
    // Handle unsupported request methods
    die("Error: Unsupported request method.");
}

// Calculate sum
$sum = intval($num1) + intval($num2);

// Print the result
echo "Sum of $num1 and $num2 is: $sum";
// while(1)
// {

// }
?>
